src/repositories/user/user.repository.ts

```diff
@@ -17,28 +18,108 @@ export class UserRepository {
   //   return acelaUser;
   // }
 
-  async findOneByEmail(email: string): Promise<User | null> {
+  async findOneByEmail(email: string): Promise<LegacyUser | null> {
     const query = { email };
     const acelaUser = await this.userModel.findOne(query);
     this.#logger.log(acelaUser);
 
     return acelaUser;
   }
 
+  async findOneBySub(sub: string) {
+    const query = { sub } satisfies Partial<LegacyUser>;
+    const authUser = await this.userModel.findOne(query);
+
+    return authUser ? authUser.toObject() : null;
+  }
+
+  async findOneByUserId(query: Pick<LegacyUser, 'user_id'>) {
+    const authUser = await this.userModel.findOne(query);
+
+    return authUser ? authUser.toObject() : null;
+  }
+
   // async insertOne() {
   //   this.userModel.create<User>({})
   // }
 
-  async verifyEmail(verifyCode: string) {
-    this.userModel.updateOne(
-      {
-        email_code: verifyCode,
-      },
-      {
-        $set: {
-          email_status: 'verified',
-        },
-      },
-    );
+  async createNewEmailUser(query: Pick<LegacyUser, 'email' | 'user_id'>) {
+    await this.userModel.create(query);
+  }
+
+  async createNewSubUser(user: Pick<LegacyUser, 'sub' | 'user_id'>) {
+    return await this.userModel.create(user);
+  }
+
+  async getLegacyLinkedHiveAccounts(user_id: string): Promise<{
+    banned: boolean;
+    linked_hiveaccounts: string[];
+    user_id: string;
+    last_hiveaccount: string;
+    _id: ObjectId;
+  }> {
+    return (
+      await this.userModel
```

Comment:
Test this without aggregate. See which is faster.

Reply:
Going to resolve this because: MongoDB aggregations are extremely likely to be more efficient than JS, there are guaranteed to be fewer requests and therefore lower latency, the cost-benefit is tiny. lets revisit if there are any problems